### PR TITLE
[RTC-208] Add server socket tests

### PR DESCRIPTION
**Pull request by mickel8**

### CODE REVIEW

File: test/jellyfish_web/integration/peer_socket_test.exs
Comment (LVala) [inline test/jellyfish_web/integration/peer_socket_test.exs:1]: As we mentioned offline, it's probably possible to move some tests from peer socket unit tests here.

File: test/jellyfish_web/integration/server_socket_test.exs
[the remark this answers is not in the document]
Comment (mickel8) [reply test/jellyfish_web/integration/server_socket_test.exs:1]: Unfortunately, we are not able to get engine endpoint pid at the moment

[the remark this answers is not in the document]
Comment (mickel8) [reply test/jellyfish_web/integration/server_socket_test.exs:1]: TBH, I don't see any benefits from using describe here. We will just introduce more nesting :thinking:

[the remark this answers is not in the document]
Comment (LVala) [reply test/jellyfish_web/integration/server_socket_test.exs:1]: I think is creates nice visual distinction between connection tests, receiving information from socket tests etc., but I don't care about it that much.

**Review by LVala — APPROVED** [review]
👍, but maybe it would be nice to add a comment somewhere why we don't test peer/component crashes (`membrane_rtc_engine` limitations).

**Pull request merged**